tests/test_image.h

```diff
@@ -222,6 +229,43 @@ TEST_CASE("[Image] Modifying pixels of an image") {
 		}
 	}
 
+	// Fill rect with color
+	{
+		const int img_width = 3;
+		const int img_height = 3;
+		Vector<Rect2> rects;
+		rects.push_back(Rect2());
+		rects.push_back(Rect2(-5, -5, 3, 3));
+		rects.push_back(Rect2(img_width, 0, 12, 12));
+		rects.push_back(Rect2(0, img_height, 12, 12));
+		rects.push_back(Rect2(img_width + 1, img_height + 2, 12, 12));
+		rects.push_back(Rect2(1, 1, 1, 1));
+		rects.push_back(Rect2(0, 1, 2, 3));
+		rects.push_back(Rect2(-5, 0, img_width + 10, 2));
+		rects.push_back(Rect2(0, -5, 2, img_height + 10));
+		rects.push_back(Rect2(-1, -1, img_width + 1, img_height + 1));
+
+		for (const Rect2 &rect : rects) {
+			Ref<Image> img = memnew(Image(img_width, img_height, false, Image::FORMAT_RGBA8));
+			CHECK_NOTHROW_MESSAGE(
+					img->fill_rect(rect, Color(1, 1, 1, 1)),
+					"fill_rect() shouldn't throw for any rect.");
+			for (int x = 0; x < img->get_width(); x++) {
```

Comment:
Nitpicking, but is there a reason you're iterating column first and not row first?

Reply:
The reason is called "copy-pasting". :)
But it's a 3x3 image, shouldn't matter.

Reply:
But I guess I can swap that when squashing.

Reply:
Isn't it normal to iterate column (width, x) first?

Reply:
Usually 2d data is stored in a contigous block of memory in a row-after-row manner (that's the case for the `Image`) so when iterating over that data it's better to iterate over the rows in the outer loop to make the iteration happen sequentially over the memory (which is cache-friendly so most likely less memory reads will be needed). If iterating over the columns first you'd keep jumping back and forth over the memory.
Maybe compiler would optimize it anyway, maybe not, no idea. It's just better/safer to iterate row first in such cases.

Reply:
I think it makes more sense logically anyway, since row first goes left to right, then top to bottom, which is how code is read, as opposed to column first, which goes top to bottom, then left to right.

Reply:
Alright, then it might be good to change indeed so we can merge :)

Reply:
@akien-mga Should I squash the commits too?

Reply:
I wouldn't squash the commits, as they represent separate changes that change different parts of the code, maybe Akien has different opinion though

Reply:
@akien-mga Just reminding about this one (before `3.5` is feature frozen :smile: ). It's good to merge unless you want the commits to be squashed.